### ISSUE TITLE
Autocomplete for Brackets

### ISSUE THREAD

**Issue by Louis9902** [latex-lsp/texlab#265]
Hey, first of all, an awesome project works really well.

But one small thing that would be very nice to have is that if you open a bracket
like `\(` for a math environment that it would be closed automatically with `\)`.

Also for completions like `\usepackage`, it would be useful if it also creates the corresponding brackets like so `\usepackage{%cursor%}`

I'm not quite sure if it fits here or to the vscode extension.

**Comment by efoerster** [comment]
Thanks for the report. I have added the missing math delimiters to the VS Code extension: https://github.com/latex-lsp/texlab-vscode/commit/5355fb8835e3b1e8e91da3307a3b5948ff4132a5. It should be available in the next release.

For the second part of your suggestion, I think that this behavior is not always wanted, for example if you need to add parameters: `\usepackage[...]{}`.

**Comment by vigoux** [comment]
I second the "argument" expansion, I think it can be really handy. Maybe having at least an option to toggle that on ?

**Comment by clason** [comment]
I agree that it'd be nice to have a configuration option (off by default) to auto-close braces on completion. (This is coming from someone using texlab with vim, so it'd be nice to have this in the texlab server, not the extension.)

**Comment by Meodoc** [comment]
Autocomplete including braces for commands like `\chapter`, `\section`, `\usepackage`, etc. would really be nice to have, since they basically always require braces afterward anyways. Braces on German keyboards are a pain to type, and for them to be included in the autocomplete would be a godsend!
